circuit/src/layered/circuit.rs

```diff
@@ -144,11 +145,33 @@ impl<C: GKRFieldConfig> Clone for Circuit<C> {
 unsafe impl<C> Send for Circuit<C> where C: GKRFieldConfig {}
 
 impl<C: GKRFieldConfig> Circuit<C> {
-    pub fn load_circuit<Cfg: GKRConfig<FieldConfig = C>>(filename: &str) -> Self {
+    // Load a circuit from a file and flatten it
+    // Used for single-thread prover or verifier
+    pub fn load_circuit_independent<Cfg: GKRConfig<FieldConfig = C>>(filename: &str) -> Self {
```

Comment:
minor: suggest renaming to `load circuit single core/process`? maybe we should stick to certain naming convention when it comes to MPI multi-procses or single process vanilla settings

Reply:
on a second thought, I suppose this method will only be used by verifier, as eventually we are deprecating MPI config from verifier, so only root verifier reads the circuit...

so a minor suggestion: do you want to align the main.rs and main_mpi.rs implementations such that they will only call the load circuit shared such that the implementation looks aligned?